src/asset_util/src/lib.rs

```diff
@@ -465,21 +448,45 @@ fn collect_assets_rec(dir: &Dir, assets: &mut Vec<Asset>) {
     }
 }
 
-/// Returns the portion of the filename after the first dot.
-/// This corresponds to the file extension for the assets handled by this canister.
-///
-/// The builtin `extension` method on `Path` does not work for file extensions with multiple dots
-/// such as `.js.gz`.
-fn file_extension<'a>(asset: &'a File) -> &'a str {
-    asset
-        .path()
-        .file_name()
-        .unwrap()
-        .to_str()
-        .unwrap()
-        .split_once('.')
-        .unwrap()
-        .1
+/// Returns the content type and the encoding type of the given file, based on the extension(s).
+/// If the text after the last dot is "gz" (i.e. this is a gzipped file), then content type
+/// is determined by the text after the second to last dot and the last dot in the file name,
+/// e.g. `ContentType::JS` for "some.gzipped.file.js.gz", and the encoding is `ContentEncoding::GZip`.
+/// Otherwise the content type is determined by the text after the last dot in the file name,
+/// and the encoding is `ContentEncoding::Identity`.
+fn content_type_and_encoding(asset: &File) -> (ContentType, ContentEncoding) {
```

Comment:
nit: you could directly pass the path here, don't think you need the actual `File` anywhere? that'll give a better idea of what's happening (inferring from the path) and will simplify your tests a tiny bit

Reply:
Done.